syntax = "proto3";

message SMSMessage {
  string mobile_number = 1;
  string message = 2;
}

message MessageID {
  string uuid = 1;
}

message MessageStatus {
  string description = 1;
  int32 code = 2;
  string ipaddress = 3;
  string response_datetime = 4;
}

message DeliveryStatus {
  string message_id = 1;
  string receipted_message_id = 2;
  string status = 3;
  string short_message = 4;
  string response_datetime = 5;
}

service SMSService {
   // Service to send SMS and get status

  // Send message
  rpc SendMessage(SMSMessage) returns (MessageID) {};
  // Get message status (sent or not)
  rpc GetMessageStatus(MessageID) returns (MessageStatus) {};
  // Get delivery report
  rpc GetDeliveryStatus(MessageID) returns (DeliveryStatus) {};
}